test/jdk/javax/swing/JTableHeader/8016524/Bug8016524.java

```diff
@@ -0,0 +1,116 @@
+import java.awt.Robot;
+import java.awt.Color;
+import javax.swing.JFrame;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+import javax.swing.UIManager;
+import javax.swing.SwingUtilities;
+import javax.swing.table.JTableHeader;
+
+public class Bug8016524 {
+
+    private static JFrame frame;
+    private static JTable table;
+    private static JScrollPane scrollableTable;
+    private static JTableHeader header;
+    private static Point point;
+
+    // added so as to get the correct pixel value of the bottom border
+    public static final int X_OFFSET = 10;
+    public static final int Y_OFFSET = 1;
+
+    public static final int FRAME_HT = 300;
+    public static final int FRAME_WT = 300;
+    public static final int TABLE_COLS = 5;
+    public static final int TABLE_ROWS = 2;
+
+    private static final int WHITE_RGB = Color.WHITE.getRGB();
+
+    public static void main(String[] args) throws Exception {
+
+        //to keep track of header dimensions
+        final int header_dim[] = new int[2];
+        Robot robot = new Robot();
+        robot.setAutoDelay(20);
+
+        try {
+            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+             throw new RuntimeException("Unsupported LookAndFeel Class: ");
+        }
+
+        SwingUtilities.invokeAndWait(() -> {
+            table = new JTable(TABLE_ROWS, TABLE_COLS);
+            scrollableTable = new JScrollPane(table);
+            frame = new JFrame();
+            frame.getContentPane().add(scrollableTable);
+            frame.setSize(FRAME_WT, FRAME_HT);
+            frame.setLocationRelativeTo(null);
+            frame.pack();
+            frame.setVisible(true);
+            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+
+            // retrieve JTableHeader coordinate position on screen (x,y)
+            point = table.getTableHeader().getLocationOnScreen();
+            // retrieve height and width of the table
+            header = table.getTableHeader();
+            header_dim[0] = header.getHeight();
+            header_dim[1] = header.getWidth();
+        });
+
+        robot.waitForIdle();
+
+        // to check mouse pointer position on screen
+        robot.mouseMove(point.x + X_OFFSET, point.y + header_dim[0] - Y_OFFSET);
+        robot.waitForIdle();
+
+        // get pixel color at lower left and lower right on the JTableHeader border
+        Color lowerLeft = robot.getPixelColor(point.x + X_OFFSET, point.y + header_dim[0] - Y_OFFSET);
+        Color lowerRight = robot.getPixelColor(point.x + header_dim[1] - X_OFFSET, point.y + header_dim[0] - Y_OFFSET);
+
+        // if pixel color is white then border not visible, throw Exception
+        if(lowerLeft.getRGB() == WHITE_RGB || lowerRight.getRGB() == WHITE_RGB)
+        {
+            throw new RuntimeException("JTableHeader Bottom Border not visible");
```

Comment:
Not sure this check is correct. If we have dark theme or the background is changed in system, then I guess the test will fail even with fix. Maybe you can set the JTable background to white explicitly.

Reply:
@prsadhuk I have explicitly set the background to white for the table as well as the header.

Reply:
@prsadhuk The fix and test case is ready for review. I have marked the testcase as mac-only.

Reply:
I see it fails for me in mac 10.15..DId you see it pass locally for you and in CI systems? Also, it should show 3 rows including row showing "A""B""C" as it is shown in windows/ubuntu, why it is showing only 1 row here?

<img width="1044" alt="Screenshot 2022-02-07 at 2 19 08 PM" src="https://user-images.githubusercontent.com/43534309/152755889-a148fd24-3b62-4e86-a3f5-5b8c0c989f77.png">

Reply:
@prsadhuk It passes in my local (mac 11.6.2) and in CI systems as well. 

![image](https://user-images.githubusercontent.com/95945681/152841823-3d41f187-6dda-4e4f-9944-eed2d161c756.png)
![image](https://user-images.githubusercontent.com/95945681/152842688-b8ed2bd2-af04-4e6f-a9b9-0001c1824386.png)

Reply:
@prsadhuk Ready for review. Changed the test case to paint JFrame to BufferedImage. The test case is marked headful though robot calls are not used, as the JFrame rendering requires it to be headful.

Reply:
I believe you could raise a awt Robot issue for not getting correct pixel color via getPixelColor(x,y) on retina display (if you are sure [JDK-8274939](https://bugs.openjdk.java.net/browse/JDK-8274939) patch didn't help)